lib/components/PoolUI.jsx

```diff
@@ -41,44 +62,49 @@ export const PoolUI = (props) => {
   const provider = walletContext.state.provider
   const usersAddress = walletContext._onboard.getState().address
 
-  const [ethBalance, setEthBalance] = useState(ethers.utils.bigNumberify(0))
-  const [poolAddresses, setPoolAddresses] = useState({
-    prizePool,
-  })
-  const [genericChainValues, setGenericChainValues] = useState({
-    loading: true,
-    tokenSymbol: 'TOKEN',
-    poolTotalSupply: '1234',
-  })
-
-  const [usersChainValues, setUsersChainValues] = useState({
-    loading: true,
-    usersTicketBalance: ethers.utils.bigNumberify(0),
-    usersTokenAllowance: ethers.utils.bigNumberify(0),
-    usersTokenBalance: ethers.utils.bigNumberify(0),
-  })
+  const [ethBalance, setEthBalance] = useAtom(ethBalanceAtom)
+  const [prizePoolType] = useAtom(prizePoolTypeAtom)
+  const [poolAddresses, setPoolAddresses] = useAtom(poolAddressesAtom)
+  const [_usersAddress, setUsersAddress] = useAtom(usersAddressAtom)
+  const [erc20Awards, setErc20Awards] = useAtom(erc20AwardsAtom)
+  const [network, setNetwork] = useAtom(networkAtom)
+  const [poolChainValues, setPoolChainValues] = useAtom(poolChainValuesAtom)
+  const [userChainValues, setUserChainValues] = useAtom(userChainValuesAtom)
 
-  useInterval(() => {
-    fetchChainData(
-      networkName,
-      usersAddress,
-      poolAddresses,
-      setPoolAddresses,
-      setGenericChainValues,
-      setUsersChainValues
-    )
-  }, 25000)
+  useEffect(() => {
+    // TODO: Probably need to reset other atoms if this changes.
```

Comment:
We should test out this scenario. It wouldn't be ideal if someone is bouncing between various prize pools and the app breaks on them 😄

Reply:
As of now the only way to change the pool address is the url, so there would be a full page refresh. I'll keep my eye on this for the UI updates though.

Reply:
The reference app is an SPA, so technically you could use the back button to go back to the index page and choose a different demo pool (if you're on Rinkeby) or enter a different pool address into that form:

<img width="499" alt="Screenshot 2020-12-03 at 15 09 55" src="https://user-images.githubusercontent.com/35717/101099508-9c07bd80-3579-11eb-9af7-a18a369b8dfb.png">

Hopefully users don't have to do full page refreshes! 😅